src/uwtools/tests/test_schemas.py

```diff
@@ -362,6 +367,94 @@ def test_schema_jedi_run_dir(jedi_prop):
     assert "88 is not of type 'string'" in errors(88)
 
 
+# make_hgrid
+
+
+def test_schema_make_hgrid():
+    config = {
+        "execution": {"executable": "make_hgrid"},
+        "run_dir": "/tmp",
+    }
+    errors = schema_validator("make-hgrid", "properties", "make_hgrid")
+    # Basic correctness:
+    assert not errors(config)
+    # All top-level keys are required:
+    for key in ("execution", "run_dir"):
+        assert f"'{key}' is a required property" in errors(with_del(config, key))
+    # Additional top-level keys are not allowed:
+    assert "Additional properties are not allowed" in errors({**config, "foo": "bar"})
+    # config needs at least the grid_type key:
+    assert "'grid_type' is a required property" in errors({"config": {}})
+
+
+def test_schema_make_hgrid_grid_type():
+    # Get errors function from schema_validator
+    errors = schema_validator("make-hgrid", "properties", "make_hgrid")
+
+    # If grid_type is "from_file", my_grid_file is required
+    config = {
```

Comment:
I regenerated `config` here each time due to issues in remapping consistently

Reply:
I'm curious about what issue you were seeing. It should be possible to DRY out these `config` definitions. If you've got a base `config`, you should be able to test an augmented version with `errors({**config, key: val})` and a diminished version with `errors(with_del(config, key[, key, ...]))`. If it doesn't work out, though, no worries -- but I might take a crack at it in #475.

Reply:
I was getting schema mismatches, although it might also have been a typo that tripped me up - I'll have another go but if I get stuck you're welcome to have at it!

Reply:
Cleaned up by defining the `grid_type` in `errors()` and bypassing the base schema entirely